packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/group-tool/GroupPalette.tsx

```diff
@@ -0,0 +1,117 @@
+import { PalettePortal } from '../PalettePortal';
+import { useHideDiagramElements } from '../../hide/useHideDiagramElements';
+import { useDiagramElementPalette } from '../useDiagramElementPalette';
+import { useFadeDiagramElements } from '../../fade/useFadeDiagramElements';
+import { PinIcon } from '../../../icons/PinIcon';
+import { usePinDiagramElements } from '../../pin/usePinDiagramElements';
+
+const usePaletteStyle = makeStyles((theme) => ({
+  palette: {
+    border: `1px solid ${theme.palette.divider}`,
+    borderRadius: '2px',
+    zIndex: 2,
+    position: 'fixed',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  paletteContent: {
+    display: 'grid',
+    gridTemplateColumns: ({ toolCount }: ContextualPaletteStyleProps) => `repeat(${Math.min(toolCount, 10)}, 36px)`,
+    gridTemplateRows: '28px',
+    gridAutoRows: '28px',
+    placeItems: 'center',
+  },
+  toolIcon: {
+    color: theme.palette.text.primary,
+  },
+}));
+
+export const GroupPalette = memo(({ x, y, isOpened }: GroupPaletteProps) => {
+  const { hideDiagramElements } = useHideDiagramElements();
+  const { fadeDiagramElements } = useFadeDiagramElements();
+  const { pinDiagramElements } = usePinDiagramElements();
+  const { hideDiagramElementPalette } = useDiagramElementPalette();
+  const [selectedElementIds, setSelectedElementIds] = useState<string[]>([]);
+
+  const onChange = useCallback(({ nodes }) => {
+    if (nodes.filter((node) => node.selected).length > 1) {
+      setSelectedElementIds(nodes.filter((node) => node.selected).map((node) => node.id));
+    } else {
+      setSelectedElementIds([]);
+    }
+  }, []);
+  useOnSelectionChange({
+    onChange,
+  });
+
+  const toolCount = 3;
+  const classes = usePaletteStyle({ toolCount });
+
+  const shouldRender = selectedElementIds.length > 1 && isOpened && x && y;
+  if (!shouldRender) {
+    return null;
+  }
+  hideDiagramElementPalette();
+
+  return (
+    <PalettePortal>
+      <Paper className={classes.palette} style={{ position: 'absolute', left: x, top: y }} data-testid="GroupPalette">
+        <div className={classes.paletteContent}>
+          <Tooltip title="Hide elements">
+            <IconButton
+              className={classes.toolIcon}
+              size="small"
+              aria-label="hide elements"
+              onClick={() => hideDiagramElements(selectedElementIds, true)}
+              data-testid="Hide-elements">
+              <VisibilityOffIcon fontSize="small" />
+            </IconButton>
+          </Tooltip>
+          <Tooltip title="Fade elements">
+            <IconButton
+              className={classes.toolIcon}
+              size="small"
+              aria-label="Fade elements"
+              onClick={() => fadeDiagramElements(selectedElementIds, true)}
+              data-testid="Fade-elements">
+              <TonalityIcon fontSize="small" />
+            </IconButton>
+          </Tooltip>
+          <Tooltip title="Pin element">
```

Comment:
Could you please rename into `Pin elements` please? (also change aria-label and data-testid)

Reply:
done